infra/pipeline.ts

```diff
@@ -0,0 +1,153 @@
+    const sourceAction = new actions.GitHubSourceAction({
+      actionName: 'GitHubSource',
+      owner: 'awslabs',
+      repo: 'amazon-ecs-local-container-endpoints',
+      oauthToken: githubAccessToken,
+      branch: 'mainline',
+      output: sourceOutput
+    });
+
+    pipeline.addStage({
+      stageName: 'Source',
+      actions: [sourceAction],
+    });
+
+    // Build stage containing build project for each architecture image
+    const buildStage = pipeline.addStage({
+      stageName: 'Build',
+    });
+
+    const verifyStage = pipeline.addStage({
+      stageName: 'Verify',
+    });
+
+    const platforms = [
+      {'arch': 'amd64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_3},
+      {'arch': 'arm64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_ARM},
+    ];
+
+    // Create build and verify project for each platform
+    for (const platform of platforms) {
+      const arch = platform['arch'];
+
+      const buildProject = new codebuild.PipelineProject(this, `BuildImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      const verifyProject = new codebuild.PipelineProject(this, `VerifyImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec_verify.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      buildProject.addToRolePolicy(new iam.PolicyStatement({
+        actions: ["ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetRepositoryPolicy",
+          "ecr:DescribeRepositories",
+          "ecr:ListImages",
+          "ecr:DescribeImages",
+          "ecr:BatchGetImage",
+          "ecr:InitiateLayerUpload",
+          "ecr:UploadLayerPart",
+          "ecr:CompleteLayerUpload",
+          "ecr:PutImage",
+          "ecr-public:*",
+          "secretsmanager:GetSecretValue",
+          "sts:GetServiceBearerToken",
+          "sts:AssumeRole",
+        ],
+        resources: ["*"]
+      }));
+
+      verifyProject.addToRolePolicy(new iam.PolicyStatement({
+        actions: ["ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetRepositoryPolicy",
+          "ecr:DescribeRepositories",
+          "ecr:ListImages",
+          "ecr:DescribeImages",
+          "ecr:BatchGetImage",
+          "ecr:InitiateLayerUpload",
+          "ecr:UploadLayerPart",
+          "ecr:CompleteLayerUpload",
+          "ecr:PutImage",
+          "secretsmanager:GetSecretValue",
```

Comment:
Do you know if it's possible to only limit this action to the `com.amazonaws.ec2.madison.dockerhub.amazon-ecs-local-container-endpoints.credentials` resource?

Reply:
See: 20f2349b480a30872863370a450be978ae987c8c